tests/test_utils/test_schemaview.py

```diff
@@ -1012,6 +1013,403 @@ def test_schemaview_enums(schema_view_no_imports: SchemaView) -> None:
 
 
+"""Tests of SchemaView range-related functions.
+
+    These tests cover range determination for slots in schemas with various combinations of default_range values and linkml:Any range declarations.
+
+    There are three schemas used in these tests:
+
+    The `range_local` (RL) schema        - based on RANGE_DATA["schema_path"]["range_local"]
+        - the default_range value in this schema is referred to as local_default;
+
+    The `range_importer` (RI) schema, which imports the range_local schema
+        - the range referred to as importer_default;
+        - the file contents are based on RANGE_DATA["schema_path"]["range_importer"]
+
+    The `range_import_importer` (RII) schema, which imports the range_importer schema
+        - the range is import_importer_default;
+        - the schema is generated as a text string, rather than being read from a file.
+
+    The range_tuple parameter specifies the default_range values for each of these schemas in the order
+    range_local, range_importer, range_import_importer.
+
+    A value of "" (EMPTY) means that there is no default_range set.
+    A value of None means that this file should not be generated.
+
+    RANGE_TUPLES contains all the combinations of range_tuple that are valid.
+
+"""
+
+RL = "range_local"
+RI = "range_importer"
+RII = "range_import_importer"
+
+RANGE_DATA = {
+    # value of the default_range in the schema
+    "default": {r: f"{r}_default" for r in [RL, RI, RII]},
+    # path to the original schema file
+    "schema_path": {r: INPUT_DIR_PATH / f"{r}.yaml" for r in [RL, RI]},
+}
+
+RLD = RANGE_DATA["default"][RL]
+RID = RANGE_DATA["default"][RI]
+RIID = RANGE_DATA["default"][RII]
+
+RANGE_TUPLES = [
+    # local schema only
+    (EMPTY, None, None),
+    (RLD, None, None),
+    # local imported by `importer` schema
+    (EMPTY, EMPTY, None),
+    (RLD, EMPTY, None),
+    (EMPTY, RID, None),
+    (RLD, RID, None),
+    # `importer` schema imported by a third schema
+    (EMPTY, EMPTY, EMPTY),
+    (EMPTY, RID, EMPTY),
+    (EMPTY, EMPTY, RIID),
+    (EMPTY, RID, RIID),
+    (RLD, EMPTY, EMPTY),
+    (RLD, RID, EMPTY),
+    (RLD, EMPTY, RIID),
+    (RLD, RID, RIID),
+]
+
+
+def save_temp_file(contents: str, filename: str, tmp_path_factory: pytest.TempPathFactory) -> Path:
+    """Save contents to a temporary file and return the path.
+
+    :param contents: the contents to save
+    :type contents: str
+    :param filename: the name of the file to create
+    :type filename: str
+    :param tmp_path_factory: pytest fixture for generating temporary paths
+    :type tmp_path_factory: pytest.TempPathFactory
+    :return: path to the generated file
+    :rtype: Path
+    """
+    tmp_file = tmp_path_factory.mktemp("test_schemaview") / filename
+    tmp_file.write_text(contents)
+    return tmp_file
+
+
+def gen_schema_name(range_tuple: tuple[str, str | None, str | None]) -> str | None:
+    """Generate a schema name from a range tuple.
```

Comment:
This naming schema allows the schema configuration and defaults to be deduced from the name alone. Wow!